modules/apps/hooks/testutils/testing_hooks.go

```diff
@@ -0,0 +1,54 @@
+package testutils
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	capabilitytypes "github.com/cosmos/cosmos-sdk/x/capability/types"
+	ibchooks "github.com/cosmos/ibc-go/v5/modules/apps/hooks"
+	channeltypes "github.com/cosmos/ibc-go/v5/modules/core/04-channel/types"
+	ibcexported "github.com/cosmos/ibc-go/v5/modules/core/exported"
+)
+
+var _ ibchooks.IBCAppHooks = TestRecvOverrides{}
+var _ ibchooks.IBCAppHooks = TestRecvBeforeAfterHooks{}
+
+type Status struct {
+	OverrideRan bool
+	BeforeRan   bool
+	AfterRan    bool
+}
+
+// Recv
+type TestRecvOverrides struct{ Status *Status }
+
+func (t TestRecvOverrides) OnRecvPacketOverride(im ibchooks.IBCMiddleware, ctx sdk.Context, packet channeltypes.Packet, relayer sdk.AccAddress) ibcexported.Acknowledgement {
+	t.Status.OverrideRan = true
+	ack := im.App.OnRecvPacket(ctx, packet, relayer)
+	return ack
+}
+
+type TestRecvBeforeAfterHooks struct{ Status *Status }
+
+func (t TestRecvBeforeAfterHooks) OnRecvPacketBeforeHook(ctx sdk.Context, packet channeltypes.Packet, relayer sdk.AccAddress) {
+	t.Status.BeforeRan = true
+}
+func (t TestRecvBeforeAfterHooks) OnRecvPacketAfterHook(ctx sdk.Context, packet channeltypes.Packet, relayer sdk.AccAddress, ack ibcexported.Acknowledgement) {
+	t.Status.AfterRan = true
+}
+
+// Send
+type TestSendOverrides struct{ Status *Status }
+
+func (t TestSendOverrides) SendPacketOverride(im ibchooks.IBCMiddleware, ctx sdk.Context, chanCap *capabilitytypes.Capability, packet ibcexported.PacketI) error {
+	t.Status.OverrideRan = true
+	err := im.SendPacket(ctx, chanCap, packet)
+	return err
+}
+
+type TestSendBeforeAfterHooks struct{ Status *Status }
```

Comment:
Both `TestSendOverrides` and `TestSendBeforeAfterHooks` are not used in the tests. Do you plan to write tests for them as well?

Reply:
I was planning to add those, but realized that that's quite hard to test because the ICS4Middlware (that should contain the hooks) is inside the IBCFeeKeeper (which doesn't export the ics4Wrapper). 

The easiest way to test this would be to add the testing hook directly on simapp, but I'm not sure if that's too intrusive. 

I've removed those for now. We will probably test all of these functions with an app that uses them (i.e.: the packet forwarding middleware)